src/java/org/apache/cassandra/utils/FBUtilities.java

```diff
@@ -133,7 +133,7 @@ public static InetAddress getLocalAddress()
             try
             {
                 localInetAddress = DatabaseDescriptor.getListenAddress() == null
-                                    ? InetAddress.getLocalHost()
+                                    ? InetAddress.getByName(InetAddress.getLocalHost().getCanonicalHostName())
```

Comment:
This generally seems reasonable.  How does this resolve in Kubernetes?

Reply:
That's a good flag. I tested this myself, and we should be ok here.

```
ptdocker@cassandra-0:/mnt/secrets/certs$ hostname -f 
cassandra-0.cassandra.data-oolong-gcp.svc.cluster.local
```
Interesting, `hostname` returns just `cassandra-0`, but the certs are much more resilient in k8s, where they do all permutations of the FQDN.

Reply:
Actually that's a bit of a lie -- in fact, this fix is also required for K8s, as there is no entry for just the hostname!